src/Organizer/Organizer.php

```diff
@@ -363,7 +324,35 @@ public function updateImage(
         );
     }
 
-    public function hasImage(UUID $imageId): bool
+    public function updateMainImage(UUID $imageId): void
+    {
+        if ($this->needsUpdateMainImage($imageId)) {
+            $this->apply(
+                new MainImageUpdated($this->actorId, $imageId->toString())
+            );
+        }
+    }
+
+    private function needsUpdateMainImage(UUID $mainImageId): bool
+    {
+        // When the organizer has no images it can't be set as main.
+        // If the organizer does not contain the main image as a normal image it can't be set as main.
+        if (!$this->hasImage($mainImageId)) {
+            return false;
+        }
+
+        // If the organizer had no main image just set it as main.
+        // Although this should be a race condition.
+        if ($this->mainImageId === null) {
+            return true;
+        }
+
+        // Only set it as main when there is a difference.
+        // This is to prevent having events in the event store with no change.
+        return !$this->mainImageId->sameAs($mainImageId);
```

Comment:
Nit: If we could store `$this->mainImageId` as `string|null` instead of `UUID|null` (in `applyImageAdded()`), this whole method could just be `return $this->hasImage($mainImageId) && $this->mainImageId !== $mainImageId;`

I'm a fan of the extra readability of these multiple steps, but one is not really necessary (only because `$this->mainImageId` is `UUID` and not just `string`), and these aggregates grow big very quickly so methods like this make them even bigger.

But functionally it works as intended and it is readable so just a nit!

Reply:
Good remark! I did tweak the usage of `mainImageId` a bit.